dss/events/handlers/index.py

```diff
@@ -126,25 +131,112 @@ def add_index_data_to_elasticsearch(bundle_key, index_data, logger) -> None:
-                                              body=json.dumps(index_data, indent=4))
-
+                                              body=json.dumps(index_data))
     except Exception as ex:
         logger.error(f"Document not indexed. Exception: {ex}  Index data: {json.dumps(index_data, indent=4)}")
+        raise
+
+
+def find_matching_subscriptions(index_data, logger):
+    percolate_document = {
+        'query': {
+            'percolate': {
+                'field': "query",
+                'document_type': DSS_ELASTICSEARCH_DOC_TYPE,
+                'document': index_data
+            }
+        }
+    }
+    subscription_ids = set()
+    for hit in scan(ElasticsearchClient.get(logger),
+                    index=DSS_ELASTICSEARCH_INDEX_NAME,
+                    query=percolate_document):
+        subscription_ids.add(hit["_id"])
+    logger.debug("Found matching subscription count: %i", len(subscription_ids))
+    return subscription_ids
+
+
+def process_notifications(bundle_id, subscription_ids, logger):
+    for subscription_id in subscription_ids:
+        try:
+            # TODO Batch this request
+            subscription = get_subscription(subscription_id, logger)
+            notify(subscription, bundle_id, logger)
+        except Exception as ex:
+            logger.error(f"Error occurred while processing subscription {subscription_id} for bundle {bundle_id}. {ex}")
+
+
+def get_subscription(subscription_id, logger):
+    subscription_query = {
+        'query': {
+            'ids': {
+                'type': DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE,
+                'values': [subscription_id]
+            }
+        }
+    }
+    response = ElasticsearchClient.get(logger).search(
+        index=DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME,
+        body=subscription_query)
+    if len(response['hits']['hits']) == 1:
+        return response['hits']['hits'][0]['_source']
+
+
+def notify(subscription, bundle_id, logger):
+    bundle_uuid, bundle_version = split_bundle_id(bundle_id)
+    transaction_id = uuid.uuid4()
+    payload = {
+        "transaction_id": transaction_id,
+        "subscription_id": subscription['subscription_id'],
+        "query": subscription['query'],
+        "match": {
+            "bundle_uuid": bundle_uuid,
+            "bundle_version": bundle_version
+        }
+    }
+    callback_url = subscription['callback_url']
+    response = requests.post(callback_url, data=payload)
+    if 200 <= response.status_code < 300:
+        logger.info(f"Successfully notified for subscription {subscription['subscription_id']}"
+                    f" for bundle {bundle_id} with transaction id {transaction_id} Code: {response.status.code}")
+    else:
+        logger.warning(f"Failed notification for subscription {subscription['subscription_id']}"
+                       f" for bundle {bundle_id} with transaction id {transaction_id} Code: {response.status.code}")
```

Comment:
`status.code` should be `status_code`. See https://travis-ci.org/HumanCellAtlas/data-store/builds/257754710 line 295